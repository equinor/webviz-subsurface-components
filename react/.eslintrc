{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module"
  },
  "extends":  [
    "plugin:react/recommended",
    "eslint:recommended",
    "plugin:prettier/recommended"
  ],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx"],
      "plugins": ["@typescript-eslint"],
      "extends": [
        "plugin:react/recommended",
        "plugin:@typescript-eslint/recommended",
        "plugin:prettier/recommended"
      ]
    }
  ],
  "settings": {
    "react": {
      "version": "detect"
    }
  },
  "env": {
    "amd": true,
    "browser": true,
    "es6": true,
    "jasmine": true,
    "node": true
  }
}
