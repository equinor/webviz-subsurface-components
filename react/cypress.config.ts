import { defineConfig } from "cypress";
import getCompareSnapshotsPlugin from "cypress-image-diff-js/dist/plugin";

export default defineConfig({
  component: {
    devServer: {
      framework: "create-react-app",
      bundler: "webpack",
    },
    video:false,
    setupNodeEvents(on, config) {
      getCompareSnapshotsPlugin(on, config),
      on('before:browser:launch', (browser = {}, launchOptions) => {
        if (browser.family === 'chromium' && browser.name !== 'electron') {
          launchOptions.args.push('--start-fullscreen')
      
          return launchOptions
        }
        if (browser.name === 'electron') {
          launchOptions.preferences.fullscreen = true
      
          return launchOptions
        }
      })
    }
  },
});
