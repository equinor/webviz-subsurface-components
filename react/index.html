<html>
    <head>
        <title>Demo application of webviz-subsurface-components</title>
    </head>
    <body>
        <div id='root'></div>
        <link rel="stylesheet" type="text/css" href="output.css">
        <script src="output.js"></script>
    </body>
</html>
