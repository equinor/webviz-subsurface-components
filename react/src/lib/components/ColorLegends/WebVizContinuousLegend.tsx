import React from "react";
import PropTypes from "prop-types";
import { ContinuousLegend } from "@emerson-eps/color-tables";
import { colorTablesArray } from "@emerson-eps/color-tables/";

interface LegendProps {
    title: string;
    min: number;
    max: number;
    position?: number[] | null;
    colorName: string;
    horizontal?: boolean | null;
    colorTables: colorTablesArray | string | undefined;
}

const ContinuousLegendWrapper: React.FC<LegendProps> = ({
    title,
    min,
    max,
    position,
    colorName,
    horizontal,
    colorTables,
}) => {
    return (
        <ContinuousLegend
            min={min}
            max={max}
            dataObjectName={title}
            position={position}
            colorName={colorName}
            horizontal={horizontal}
            colorTables={colorTables}
        />
    );
};

ContinuousLegendWrapper.propTypes = {
    title: PropTypes.string.isRequired,
    min: PropTypes.number.isRequired,
    max: PropTypes.number.isRequired,
    position: PropTypes.arrayOf(PropTypes.number.isRequired),
    colorName: PropTypes.string.isRequired,
    horizontal: PropTypes.bool,
    colorTables: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
};

export default ContinuousLegendWrapper;
