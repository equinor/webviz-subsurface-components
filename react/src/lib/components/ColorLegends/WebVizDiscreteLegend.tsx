import React from "react";
import PropTypes from "prop-types";
import { DiscreteColorLegend } from "@emerson-eps/color-tables";
import { colorTablesArray } from "@emerson-eps/color-tables/";

interface LegendProps {
    discreteData: { objects: Record<string, [number[], number]> };
    title: string;
    position?: number[] | null;
    colorName: string;
    colorTables: colorTablesArray | string | undefined;
    horizontal?: boolean | null;
}

const DiscreteLegendWrapper: React.FC<LegendProps> = ({
    discreteData,
    title,
    position,
    colorName,
    colorTables,
    horizontal,
}) => {
    return (
        <DiscreteColorLegend
            discreteData={discreteData}
            dataObjectName={title}
            position={position}
            colorName={colorName}
            colorTables={colorTables}
            horizontal={horizontal}
        />
    );
};

DiscreteLegendWrapper.propTypes = {
    discreteData: PropTypes.any.isRequired,
    title: PropTypes.string.isRequired,
    position: PropTypes.arrayOf(PropTypes.number.isRequired),
    colorName: PropTypes.string.isRequired,
    colorTables: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
    horizontal: PropTypes.bool,
};

export default DiscreteLegendWrapper;
