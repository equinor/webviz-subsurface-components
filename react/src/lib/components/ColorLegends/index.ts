import ContinuousLegendWrapper from "./WebVizContinuousLegend";
import DiscreteLegendWrapper from "./WebVizDiscreteLegend";

export { ContinuousLegendWrapper, DiscreteLegendWrapper };
