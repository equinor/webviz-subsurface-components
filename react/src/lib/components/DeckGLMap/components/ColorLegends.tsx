import React from "react";
import { ExtendedLayer } from "../layers/utils/layerTools";
import ColorLegend from "./ColorLegend";

interface ColorLegendsProps {
    // Pass additional css style to the parent color legend container
    cssStyle?: Record<string, unknown> | null;
    horizontal?: boolean | null;
    layers: ExtendedLayer<unknown>[];
}

// Todo: Adapt it for other layers too
const ColorLegends: React.FC<ColorLegendsProps> = ({
    cssStyle,
    horizontal,
    layers,
}: ColorLegendsProps) => {
    if (layers.length == 0) return null;
    return (
        <div
            style={{
                position: "absolute",
                display: "flex",
                zIndex: 999,
                ...cssStyle,
            }}
        >
            {layers.map((layer, index) => (
                <ColorLegend
                    layer={layer}
                    horizontal={horizontal}
                    key={index}
                />
            ))}
        </div>
    );
};

export default ColorLegends;
