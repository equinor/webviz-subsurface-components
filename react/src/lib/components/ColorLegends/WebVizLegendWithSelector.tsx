import React from "react";
import PropTypes from "prop-types";
import { ContinuousLegend } from "../DeckGLMap/components/ContinuousLegend";
import {colorTablesArray} from "@emerson-eps/color-tables/src/component/ColorTableTypes";

interface LegendProps {
    title: string;
    min: number;
    max: number;
    position?: number[] | null;
    colorName: string;
    horizontal?: boolean | null;
    colorTables: colorTablesArray | string;
}

const ContinuousLegendWrapper: React.FC<LegendProps> = ({
    title,
    min,
    max,
    position,
    colorName,
    horizontal,
    colorTables,
}) => {
    return (
        <ContinuousLegend
            min={min}
            max={max}
            dataObjectName={title}
            position={position}
            colorName={colorName}
            horizontal={horizontal}
            colorTables={colorTables}
        />
    );
};

ContinuousLegendWrapper.propTypes = {
    title: PropTypes.string.isRequired,
    min: PropTypes.number.isRequired,
    max: PropTypes.number.isRequired,
    position: PropTypes.arrayOf(PropTypes.number.isRequired),
    colorName: PropTypes.string.isRequired,
    horizontal: PropTypes.bool,
    colorTables: PropTypes.any,
};

export default ContinuousLegendWrapper;