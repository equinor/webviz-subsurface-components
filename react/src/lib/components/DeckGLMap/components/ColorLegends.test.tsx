/* eslint-disable @typescript-eslint/no-var-requires */
import { render } from "@testing-library/react";
import "jest-styled-components";
import "@testing-library/jest-dom";
import React from "react";
import ColorLegends from "./ColorLegends";
import { Layer } from "@deck.gl/core/typed";
import {
    getLayersInViewport,
    getLayersWithDefaultProps,
} from "../layers/utils/layerTools";

const exampleData = require("../../../../demo/example-data/deckgl-map.json");
const layers = getLayersWithDefaultProps(exampleData[0].layers);
import { colorTables } from "@emerson-eps/color-tables";

describe("Test Color Legend", () => {
    it("snapshot test", () => {
        const { container } = render(
            <ColorLegends
                layers={
                    getLayersInViewport(layers, [
                        "colormap-layer",
                        "wells-layer",
                    ]) as Layer<unknown>[]
                }
                colorTables={colorTables}
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
    it("snapshot test with props", () => {
        const { container } = render(
            <ColorLegends
                layers={
                    getLayersInViewport(layers, [
                        "colormap-layer",
                        "wells-layer",
                    ]) as Layer<unknown>[]
                }
                horizontal={true}
                colorTables={colorTables}
                reverseRange={true}
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
    it("snapshot test if layers=0", () => {
        const { container } = render(
            <ColorLegends
                layers={[]}
                horizontal={true}
                colorTables={colorTables}
                reverseRange={true}
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
