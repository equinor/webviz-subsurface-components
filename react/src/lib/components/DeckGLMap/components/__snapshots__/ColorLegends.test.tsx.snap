// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Color Legend snapshot test 1`] = `null`;
